File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppJava/app/src/main/java/pl/udu/uwr/pum/verynobleappjava/adapters/laureatenobelprizes/NobelPrizeComparator.java
package pl.udu.uwr.pum.verynobleappjava.adapters.laureatenobelprizes;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.nobelprize.NobelPrize;

public class NobelPrizeComparator extends DiffUtil.ItemCallback<NobelPrize> {
    @Override
    public boolean areItemsTheSame(@NonNull NobelPrize oldItem, @NonNull NobelPrize newItem) {
        return newItem == oldItem;
    }

    @Override
    public boolean areContentsTheSame(@NonNull NobelPrize oldItem, @NonNull NobelPrize newItem) {
        return newItem.awardYear.equals(oldItem.awardYear);
    }
}

File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppKotlin/app/src/main/java/pl/udu/uwr/pum/verynobleappkotlin/adapters/nobelprize/NobelPrizeComparator.kt
package pl.udu.uwr.pum.verynobleappkotlin.adapters.nobelprize

import androidx.recyclerview.widget.DiffUtil
import pl.udu.uwr.pum.verynobleappkotlin.data.nobelprizeresponse.nobelprize.NobelPrize

class NobelPrizeComparator : DiffUtil.ItemCallback<NobelPrize>() {
    override fun areItemsTheSame(oldItem: NobelPrize, newItem: NobelPrize): Boolean {
        return oldItem.awardYear == newItem.awardYear
    }

    override fun areContentsTheSame(oldItem: NobelPrize, newItem: NobelPrize): Boolean {
        return oldItem == newItem
    }
}